src/Tests/TemplateTests.NServiceBusWindowsService.approved.txt

```diff
@@ -1,4 +1,110 @@
+
+                // TODO: ensure the most appropriate serializer is chosen
+                // https://docs.particular.net/nservicebus/serialization/
+                endpointConfiguration.UseSerialization<NewtonsoftSerializer>();
+
+                // TODO: remove this condition after choosing a transport, persistence and deployment method suitable for production
+                if (Environment.UserInteractive && Debugger.IsAttached)
+                {
+                    // TODO: choose a durable transport for production
+                    // https://docs.particular.net/transports/
+                    var transportExtensions = endpointConfiguration.UseTransport<LearningTransport>();
+
+                    // TODO: choose a durable persistence for production
+                    // https://docs.particular.net/persistence/
+                    endpointConfiguration.UsePersistence<LearningPersistence>();
+
+                    // TODO: create a script for deployment to production
+                    endpointConfiguration.EnableInstallers();
+                }
+
+                // TODO: perform any futher start up operations before or after starting the endpoint
+                endpoint = await Endpoint.Start(endpointConfiguration);
+            }
+            catch (Exception ex)
+            {
+                await FailFast("Failed to start.", ex);
+            }
+        }
+
+        public async Task Stop()
+        {
+            try
+            {
+                // TODO: perform any futher shutdown operations before or after stopping the endpoint
+                await endpoint?.Stop();
+            }
+            catch (Exception ex)
+            {
+                await FailFast("Failed to stop correctly.", ex);
+            }
+        }
+
+        async Task OnCriticalError(ICriticalErrorContext context)
+        {
+            // TODO: decide if stopping the endpoint and exiting the process is the best response to a critical error
+            // https://docs.particular.net/nservicebus/hosting/critical-errors
+            // and consider setting up service recovery
+            // https://docs.particular.net/nservicebus/hosting/windows-service#installation-restart-recovery
+            try
+            {
+                await context.Stop();
+            }
+            finally
+            {
+                await FailFast($"Critical error, shutting down: {context.Error}", context.Exception);
+            }
+        }
+
+        async Task FailFast(string message, Exception exception)
+        {
+            try
+            {
+                log.Fatal(message, exception);
+
+                // TODO: when using an external logging framework it is important to flush any pending entries prior to calling FailFast
+                // https://docs.particular.net/nservicebus/hosting/critical-errors#when-to-override-the-default-critical-error-action
+                await Task.CompletedTask;
```

Comment:
Why are we awaiting a `Task.CompletedTask` here? Why is the method async? Why not just return void here?

Reply:
Yea lets make this void.

One argument could be to prep for loggers becoming async but https://github.com/aspnet/Logging/issues/444 is a good description on why that's not that likely